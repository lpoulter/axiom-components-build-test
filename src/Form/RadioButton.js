import PropTypes from 'prop-types';
import React, { Component } from 'react';
import Validate from '../Validation/Validate';
import ChedioButtox from './ChedioButtox';

export default class RadioButton extends Component {
  static propTypes = {
    /** Checked state of the input */
    checked: PropTypes.bool,
    /** Label that is placed next to the input */
    children: PropTypes.node,
    /** Disables interactions and applies styling */
    disabled: PropTypes.bool,
    /** See Validate error */
    error: PropTypes.func,
    /** Applies styling to indicate the users input was invalid */
    invalid: PropTypes.bool,
    /** Name of the group the input belongs to */
    name: PropTypes.string,
    /** See Validate[patterns] */
    patterns: PropTypes.array,
    /** See Validate[required] */
    required: PropTypes.bool,
    /** Text that is displayed on hover */
    title: PropTypes.string,
  };

  render() {
    const {
      checked,
      children,
      disabled,
      error,
      invalid,
      name,
      patterns,
      required,
      ...rest
    } = this.props;

    return (
      <Validate
          error={ error }
          patterns={ patterns }
          required={ required }
          value={ checked }>
        { (isValid) =>
          <ChedioButtox
              { ...rest }
              checked={ checked }
              className="ax-radio"
              disabled={ disabled }
              inputType="radio"
              invalid={ invalid || isValid === false }
              name={ name }>
            { children }
          </ChedioButtox>
        }
      </Validate>
    );
  }
}
