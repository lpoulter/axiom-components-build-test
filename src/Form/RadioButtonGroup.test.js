import React from 'react';
import renderer from 'react-test-renderer';
import RadioButton from './RadioButton';
import RadioButtonGroup from './RadioButtonGroup';

const getComponent = (props = {}) =>
  renderer.create(
    <RadioButtonGroup { ...props }>
      <RadioButton>Lorem ipsum</RadioButton>
    </RadioButtonGroup>
  );

describe('RadioButton', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
