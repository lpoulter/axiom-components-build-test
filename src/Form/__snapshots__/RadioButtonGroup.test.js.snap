// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton renders with defaultProps 1`] = `
<div
  className="ax-chedio-buttox-group"
>
  <label
    className="ax-radio ax-space--x2"
  >
    <input
      className="ax-radio__input"
      type="radio"
    />
    <span
      className="ax-radio__indicator"
    />
    <span
      className="ax-radio__label"
    >
      Lorem ipsum
    </span>
  </label>
</div>
`;
